python-checks/src/main/java/org/sonar/python/checks/DeadStoreCheck.java

```diff
@@ -95,13 +106,47 @@ private static void raiseIssue(SubscriptionContext ctx, DeadStoreUtils.Unnecessa
       issue = ctx.addIssue(element.firstToken(), lastRelevantToken, message);
     }
 
+    unnecessaryAssignment.symbol.usages().stream()
+      .filter(Usage::isBindingUsage)
+      .map(Usage::tree)
+      // skip initial issue binding
+      .filter(tree -> tree != element && TreeUtils.firstAncestor(tree, parent -> parent == element) == null)
+      //skip assignments before
+      .filter(tree -> getTreeByPositionComparator().compare(tree, element) == 1)
```

Comment:
I see why we need this, but I think we're missing a test case for it (if I remove the line, all tests still succeed). Can you add a small test covering this case?
Also, IntelliJ suggests changing the comparison to `> 0` instead.